formats/json/commonTest/src/kotlinx/serialization/features/inline/InlineClassesCompleteTest.kt

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2017-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+@file:Suppress("INLINE_CLASSES_NOT_SUPPORTED", "SERIALIZER_NOT_FOUND")
+
+package kotlinx.serialization.features.inline
+
+import kotlinx.serialization.*
+import kotlinx.serialization.Box
+import kotlinx.serialization.test.*
+import kotlin.test.*
+
+@Serializable
+inline class MyInt(val i: Int)
+
+@Serializable
+inline class NullableMyInt(val i: Int?)
+
+@Serializable
+inline class OverSerializable(val s: IntData)
+
+@Serializable
+inline class OverSerializableNullable(val s: IntData?)
+
+@Serializable
+inline class WithT<T>(val t: Box<T>)
+
+@Serializable
+inline class WithTNullable<T>(val t: Box<T>?)
+
+@Serializable
+data class WithAll(
+    val myInt: MyInt, // I
+    val myIntNullable: MyInt?, // LMyInt;
+    val nullableMyInt: NullableMyInt, //Ljava/lang/Integer;
+    val nullableMyIntNullable: NullableMyInt?, // LNullableMyInt
+    val overSerializable: OverSerializable, // LIntData;
+    val overSerializableNullable: OverSerializable?, // LIntData;
+    val nullableOverSerializable: OverSerializableNullable, // LIntData;
+    val nullableOverSerializableNullable: OverSerializableNullable?, // LOverSerializableNullable;
+    val withT: WithT<Int>, // LBox;
+    val withTNullable: WithT<Int>?, // LBox
+    val withNullableTNullable: WithT<Int?>?, // LBox;
+    val withTNullableTNullable: WithTNullable<Int?>? // LWithTNullable;
+)
+
+@Serializable
+data class WithGenerics(
```

Comment:
Here the scope of the test classes can also be reduced to `InlineClassesCompleteTest`